wireapp/wire-android-reloaded
# fix: drop sync after receiving push notifications

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

App doesn't receive notifications after a while in the Background.


### Causes

Kalium has some issues that are fixed in the PRs mentioned above.
It also introduces `ConnectionPolicy` so we can make it stop Sync if the app is in the background.

### Solutions

`ConnectionPolicyManager` to observe the app lifecycle.
If the UI was never initialised, then set all sessions to `DISCONNECT_AFTER_PENDING_EVENTS`.

If the UI was initialised, then set the current active session to `KEEP_ALIVE`.

Just observing Background/Foreground (with onPause and onResume) can lead to sync starting and stopping unnecessarily as the user just opens another app for a couple of seconds.

### Dependencies

Needs releases with:

- https://github.com/wireapp/kalium/pull/707
- https://github.com/wireapp/kalium/pull/706

### Testing

#### Test Coverage (Optional)


#### How to Test

Open the app, verify that you have Sync going on.
Force-close the application, send yourself a message.
App should receive a notifications and keep receiving in the future (delay between notifications caused by SlowSync in the background).

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
